modules/self-managed-node-group/main.tf

```
@@ -255,6 +259,8 @@ locals {
   launch_template_name = try(aws_launch_template.this[0].name, var.launch_template_name)
   # Change order to allow users to set version priority before using defaults
   launch_template_version = coalesce(var.launch_template_version, try(aws_launch_template.this[0].default_version, "$Default"))
+
+  asg_default_tags = try(var.asg_use_default_tags, false) ? merge(data.aws_default_tags.current.tags, var.tags) : var.tags
```

This won't work like this, it will need to be:
```suggestion
  default_tags = var.use_default_tags ? merge(data.aws_default_tags.current.tags, var.tags) : var.tags
```

---

I disagree that it won't work, as it seemed to work fine when I tested it.  Also I copied the `try(var.my_var, false) ? "Do true" : "Do false"` pattern from the other Boolean vars in the `node_groups.tf` file.

---

`try()` works on "errors" and returns the first result that does not throw an error https://www.terraform.io/language/functions/try

There is no error here so no matter what happens, it will always take the variable listed first in the sequence so there is no point in listing it this way

---

Yep, you're 100% correct, and as I was grabbing my laptop to look into the `launch_template` tags, I realized the mistake I made.   In `node_groups.tf` the `try()` works because it's attempting to access a key on another var,  but you're right that it won't work in the child modules.  I'll push a fix shortly.

---

I fixed the ternary logic.